pxr/usd/sdf/childrenView.h

```diff
@@ -76,27 +75,75 @@ template <typename _Owner, typename _InnerIterator, typename _DummyPredicate>
 class Sdf_ChildrenViewTraits {
 private:
 
-    // Internal predicate object which will be passed to the filter
-    // iterator. This just calls through to the owner's predicate.
-    class _Predicate {
+    // Owner's predicate object will be used by the filter iterator.
+    // In C++20, consider using the ranges library to simplify this
+    class _FilterIterator {
     public:
-        typedef typename _Owner::value_type value_type;
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = typename _InnerIterator::value_type;
+        using reference = typename _InnerIterator::reference;
+        using pointer = typename _InnerIterator::pointer;
+        using difference_type = typename _InnerIterator::difference_type;
+
+        _FilterIterator() = default;
```

Comment:
The body of all three of these should be on a new line like you did for all the rest of the functions.